src/overlay/PeerManager.cpp

```diff
@@ -197,6 +157,25 @@ PeerManager::loadRandomPeers(PeerQuery const& query, size_t size)
     return result;
 }
 
+std::vector<PeerBareAddress>
+PeerManager::getPeersToSend(size_t size, PeerBareAddress const& address)
+{
+    auto keep = [&](PeerBareAddress const& pba) {
+        return !pba.isPrivate() && pba != address;
+    };
+
+    auto peers = mOutboundPeersToSend->getRandomPeers(size, keep);
```

Comment:
This will go with this part of query builder:

```
    if (query.mRequireOutbound)
    {
        if (*query.mRequireOutbound)
        {
            conditions.push_back("type != :inboundType");
        }
        else
        {
            conditions.push_back("type = :inboundType");
        }
    }
```

So either inbound (mInboundPeersToSend) or outbound+preferred (mOutboundPeersToSend).

Do you want me to extract third one just for preferred peers?

Reply:
Oh I see, I didn't realize that the query was done like this. Having those two filters on the same field is actually quite confusing and hard to read (looking at `RandomPeerSource::maxFailures`).

Why not change queries to be something like this:
```c++
enum class PeerTypeFilter {
    INBOUND_ONLY,
    ANY_OUTBOUND,
    PREFERRED_ONLY
};

struct PeerQuery
{
    bool mUseNextAttempt;
    int mMaxNumFailures;
    PeerTypeFilter mPeerTypeFilter;
};
```
That way, call sites like `maxFailures` look like this
```c++
PeerQuery
RandomPeerSource::maxFailures(int maxFailures, bool requireOutbound)
{
    return {false, maxFailures, requireOutbound?PeerTypeFilter::ANY_OUTBOUND:PeerTypeFilter::INBOUND_ONLY};
}
```